synapse/handlers/pagination.py

```diff
@@ -399,23 +435,49 @@ def get_delete_ids_by_room(self, room_id: str) -> Optional[StrCollection]:
         """
         return self._delete_by_room.get(room_id)
 
-    async def purge_room(self, room_id: str, force: bool = False) -> None:
+    async def purge_room(
+        self,
+        room_id: str,
+        delete_id: str,
+        force: bool = False,
+        shutdown_response: Optional[ShutdownRoomResponse] = None,
+    ) -> None:
         """Purge the given room from the database.
-        This function is part the delete room v1 API.
 
         Args:
             room_id: room to be purged
+            delete_id: the delete ID for this purge
             force: set true to skip checking for joined users.
+            shutdown_response: optional response coming from the shutdown phase
         """
+        logger.info("starting purge room_id %s", room_id)
+
         async with self.pagination_lock.write(room_id):
             # first check that we have no users in this room
             if not force:
                 joined = await self.store.is_host_joined(room_id, self._server_name)
                 if joined:
                     raise SynapseError(400, "Users are still joined to this room")
```

Comment:
Not a new thing from this PR but even if we're using `force`, it still seems useful to check if our server is joined and log a warning.

What are the repercussions of forcing with users still joined to the room?

Reply:
I added a log line.

For the consequence... I have no idea sorry.